### PR TITLE
Filter fix when alternating all <-> any

### LINKED ISSUE #3193: [Bug]: any/all on filters are broken
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

1. create a new custom report
2. click the filter button and add a category filter of "contains t"
3. click the filter button and add a category filter of "contains a"
4. note that it shows both, with "all" next to the filters
5. change "all" to "any"

Observed:
image

and the filters still behave as if "all" are selected

Expected:
it says "any" and the filters switch to "any" mode

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Chrome

### Operating System

Mac OSX

### PR DESCRIPTION
**Pull request by lelemm**
fixes #3193

### CODE REVIEW

File: packages/desktop-client/src/components/modals/EditRule.jsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ export function FieldSelect({ fields, style, value, onChange }) {
         bare
         options={fields}
         value={value}
-        onChange={value => onChange('field', value)}
+        onChange={value => onChange(value)}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/EditRule.jsx:97]: **🥜 nitpick:** ‏
```suggestion
        onChange={onChange}
```

**Review by MatissJanis — APPROVED** [review]
Thanks! Works as expected! Will wait for another review before merging.

Tested:
- account page
- custom reports
- generic reports
- rules

**Comment by lelemm** [comment]
saw the mergefreeze lock @MatissJanis , don't this must get into the next release?

**Comment by MatissJanis** [comment]
Definitely want this for the upcoming release. Just waiting for 1 more maintainer approval before merging.

**Review by jfdoming — APPROVED** [review]
looks good, thanks for the fix! could be nice to add tests for this at a later date

**Pull request merged**